### PR TITLE
Added matplotlib helpers

### PR DESCRIPTION
**Pull request by andyneff**
- Added custom template function for normxcorr2
- Docs compiling

### CODE REVIEW

File: python/vsi/tools/mpl.py [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+  c_colored = scalarMap.to_rgba(c)
+
+  surf = ax.plot_surface(x, y, z, facecolors=c_colored, **kwargs)
+
+  return surf
+
+
+class SimpleBubblePicker:
+  '''
+  Simple class to add add a picker with your own text function
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:205]:
```suggestion
  Simple class to add a picker with your own text function
```

File: python/vsi/tools/mpl.py [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+
+  Currently supports plots (Line2D), scatter (PathCollection), images
+  (AxesImage) and PatchCollection
+
+  With :class:`SimpleBubblePicker`, comes with a default text function that
+  will display basic information about any point you click.
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:211]:
```suggestion
  :class:`SimpleBubblePicker` comes with a default text function that will
  display basic information about any point you click.
```

File: python/vsi/tools/mpl.py [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+               **kwargs):
+    '''
+    Initialize a :class:`SimpleBubblePicker` for a specific figure.
+
+    Parameters
+    ----------
+        fig : :class:`matplotlib.figure.Figure`, optional
+            Default to using the :func:`matplotlib.pyplot.gcf`
+        text_function : :class:`py::function`, optional
+            A Defaults to :func:`SimpleBubblePicker.default_text`
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:225]:
```suggestion
            Defaults to :func:`SimpleBubblePicker.default_text`
```

File: python/vsi/tools/mpl.py [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+        x_offset : float, optional
+        y_offset : float, optional
+            Offset location of bubble from actual point
+        offset_units : 'str', optional
+            Default: ``dots``
+        **kwargs :
+            Additional argument to be passed to text initializer
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:232]:
```suggestion
            Additional keyword arguments to be passed to text initializer
```

File: python/vsi/tools/mpl.py [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+      ``ax1`` and returns ``(x0, x1, y0, y1)`` for ``ax2``. Return ``None`` to
+      indicate that no update to ``ax2`` should be performed.
+  backward : Callable
+      Same as ``forward`` except maps ``ax2`` to ``ax1``
+  sharex : :class:`bool`, optional
+      Should the x axis be shared
+  sharey : :class:`bool`, optional
+      Should the y axis be shared
+  **kwargs
+      Arbitrary keyword arguments passed to ``forward`` and ``backward``
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:39]:
```suggestion
      Additional keyword arguments passed to ``forward`` and ``backward``
```

File: python/vsi/tools/mpl.py [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+  img : :class:`numpy.ndarray`
+      Image array
+  origin : tuple
+      The origin of the chip to load, in y, x, order
+  size :
+      The size of the chip to load, in y, x, order
+  *args :
+      Additional parameters passed to :func:`matplotlib.pyplot.imshow`
+  **kwargs
+      Additional parameters passed to :func:`matplotlib.pyplot.imshow`
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:102]:
```suggestion
      Additional keyword arguments passed to :func:`matplotlib.pyplot.imshow`
```

File: python/vsi/tools/mpl.py [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+  raster : :class:`rasterio.io.DatasetReader`
+      rasterio dataset object
+  origin :
+      The origin of the chip to load, in y, x, order
+  size :
+      The size of the chip to load, in y, x, order
+  *args :
+      Additional parameters passed to :func:`matplotlib.pyplot.imshow`
+  **kwargs
+      Additional parameters passed to :func:`matplotlib.pyplot.imshow`
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:135]:
```suggestion
      Additional keyword arguments passed to :func:`matplotlib.pyplot.imshow`
```

File: python/vsi/tools/mpl.py [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+      The axes to draw on. Default: gca()
+  x : :class:`numpy.ndarray`, optional
+      The x coordinate used to draw. Default uses :func:`numpy.meshgrid`
+  y : :class:`numpy.ndarray`, optional
+      The y coordinate used to draw. Default uses :func:`numpy.meshgrid`
+  c : :class:`numpy.ndarray`, optional
+      A custom array that is fed into the colormap for coloring. Default uses
+      ``z``
+  **kwargs : dict
+      Additional parameters passed to :meth:`mpl_toolkits.mplot3d.axes3d.Axes3D.plot_surface`
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:168]:
```suggestion
      Additional keyword arguments passed to :meth:`mpl_toolkits.mplot3d.axes3d.Axes3D.plot_surface`
```

**Review by drewgilliam — APPROVED** [review]

File: python/vsi/tools/mpl.py [hunk 9]
```diff
@@ -0,0 +1,429 @@
...
+        self.x, self.y = event.artist.get_position()
+        self.text_x = self.x
+        self.text_y = self.y
+      elif isinstance(event.artist, mpl.patches.Rectangle):
+        self.bbox = event.artist.get_bbox()
+        # self.x, self.y = event.artist.get_bbox()
+        self.text_x = (self.bbox.x0 + self.bbox.x1)/2
+        self.text_y = (self.bbox.y0 + self.bbox.y1)/2
+      else:
+        plt.gca().set_title('oops')
```
Comment (scott-vsi) [inline python/vsi/tools/mpl.py:379]: Should this raise an Unimplemented Exception? or just set the title to the empty string?

Comment (andyneff) [reply python/vsi/tools/mpl.py:379]: Exceptions don't work here. They are unobservable. So I picks something that is observable.

**Pull request merged**